src/test/java/org/corfudb/infrastructure/ManagementServerTest.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.infrastructure;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Checks the various services and messages handled by the
+ * Management Server.
+ * <p>
+ * Created by zlokhandwala on 11/2/16.
+ */
+public class ManagementServerTest extends AbstractServerTest {
+
+    @Override
+    public ManagementServer getDefaultServer() {
+        ServerContext serverContext = new ServerContextBuilder()
+                .setSingle(false)
+                .setServerRouter(getRouter())
+                .build();
+        return new ManagementServer(serverContext, new LayoutServer(serverContext));
+    }
+
+    /**
+     * Testing the status of the failure detector and shutdown functionality.
+     */
+    @Test
+    public void checkFailureDetectorStatus() {
```

Comment:
I don't see how this test is effective. The isShutdown flag is switched right after the method invocation, so essentially this test only verifies that the constructor and shutdown method have been called, nothing more. It also doesnt test the side effects of the shutdown and start operation.